src/static/js/earn.js

```diff
@@ -1860,140 +3414,625 @@ async function main() {
+        break
+      case 'claimPool7':
+        claimPool7()
+        break
+      case 'claimPool8':
+        claimPool8()
+        break
+      default:
+        alert('Oops something went wrong. Try refreshing the page.')
+    }
+  })
+
+  $('.approveBtn').unbind('click')
+  $('.approveBtn').click(function() {
+    let fn = $(this).attr('data-btn')
+    switch (fn) {
+      case 'approveSPGLSUSHI':
+        approveSPGLSUSHI()
+        break
+      case 'approveSNOB':
+        approveSNOB()
+        break
+      case 'approveSPGLPNG':
+        approveSPGLPNG()
+        break
+      case 'approveSPGLETH':
+        approveSPGLETH()
+        break
+      case 'approveSPGLUSDT':
+        approveSPGLUSDT()
+        break
+      case 'approveSPGLLINK':
+        approveSPGLLINK()
+        break
+      case 'approveS3D':
+        approveS3D()
+        break
+      case 'approveS3F':
+        approveS3F()
+        break
+      default:
+        alert('Oops something went wrong. Try refreshing the page.')
+    }
+  })
+
+  $('.stakeBtn').unbind('click')
+  $('.stakeBtn').click(function() {
+    let fn = $(this).attr('data-btn')
+    switch (fn) {
+      case 'stakeSPGLSUSHI':
+        stakeSPGLSUSHI()
+        break
+      case 'stakeSNOB':
+        stakeSNOB()
+        break
+      case 'stakeSPGLPNG':
+        stakeSPGLPNG()
+        break
+      case 'stakeSPGLETH':
+        stakeSPGLETH()
+        break
+      case 'stakeSPGLUSDT':
+        stakeSPGLUSDT()
+        break
+      case 'stakeSPGLLINK':
+        stakeSPGLLINK()
+        break
+      case 'stakeS3D':
+        stakeS3D()
+        break
+      case 'stakeS3F':
+        stakeS3F()
+        break
+      default:
+        alert('Oops something went wrong. Try refreshing the page.')
+    }
   })
 
-  $(".unstakeBtn").click(function(){
-    let fn = $(this).attr("data-btn");
-    switch (fn) {
-      case 'withdrawPool1':
-        withdrawPool1();
-        break;
-      case 'withdrawPool2':
-        withdrawPool2();
-        break;
-      case 'withdrawPool3':
-        withdrawPool3();
-        break;
-      case 'withdrawPool4':
-        withdrawPool4();
-        break;
-      case 'withdrawPool5':
-        withdrawPool5();
-        break;
-      case 'withdrawPool6':
-        withdrawPool6();
-        break;
-      case 'withdrawPool7':
-        withdrawPool7();
-        break;
-      case 'withdrawPool8':
-        withdrawPool8();
-        break;
-      default:
-        alert('Oops something went wrong. Try refreshing the page.');
+  function stakeUnstake(amount, stake, st) {
+    return `<div class="col-sm-12 col-md-3 align-items-center text-center snob-tvl pb-10 pb-md-0">
+    <p class="m-0 font-size-12"><ion-icon name="pie-chart-outline"></ion-icon> You have</p>
+    <p class="m-0 font-size-16 font-weight-regular">${amount} ${st ? st : 'sPGL'} </p>
+    <p class="m-0 font-size-12">(Available to ${stake ? 'Stake' : 'Unstake'}) </p>
+    </div>`
+  }
+
+  function poolS3F(options) {
```

Comment:
The **poolS3F** is an original function. It is long, but I don't know the business well enough to change.